hudi-common/src/main/java/org/apache/hudi/common/table/log/BaseHoodieMergedLogRecordScanner.java

```diff
@@ -217,11 +229,15 @@ protected void processNextDeletedRecord(DeleteRecord deleteRecord) {
     // Put the DELETE record
     if (recordType == HoodieRecord.HoodieRecordType.AVRO) {
       records.put((K) key, SpillableMapUtils.generateEmptyPayload(key,
-          deleteRecord.getPartitionPath(), deleteRecord.getOrderingValue(), getPayloadClassFQN()));
+          deleteRecord.getPartitionPath(), deleteOrderingVal, getPayloadClassFQN()));
     } else {
-      HoodieEmptyRecord record = new HoodieEmptyRecord<>(new HoodieKey(key, deleteRecord.getPartitionPath()), null, deleteRecord.getOrderingValue(), recordType);
+      HoodieEmptyRecord record = new HoodieEmptyRecord<>(new HoodieKey(key, deleteRecord.getPartitionPath()), null, deleteOrderingVal, recordType);
       records.put((K) key, record);
     }
+
+    if (deleteRecord.getOrderingValue() == null) {
+      records.get((K) key).addMetadata(HoodieReaderContext.DELETE_FOUND_WITHOUT_ORDERING_VALUE, "true");
+    }
```

Comment:
If we always assume that `(integer) 0` represents natural ordering, is `DELETE_FOUND_WITHOUT_ORDERING_VALUE` still necessary?

Reply:
I think so since hoodie records can have natural ordering (though I don't know when a user want to use natural order). especially true, when we want to propagate this delete information through non-delete records.